src/sqlfluff/rules/L009.py

```diff
@@ -85,44 +86,39 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         if not self.is_final_segment(context):
             return None
 
-        # Include current segment for complete stack.
-        complete_stack: List[BaseSegment] = list(context.raw_stack)
-        complete_stack.append(context.segment)
-
-        # Iterate backwards over complete stack to find
-        # last non-newline/whitespace/Dedent segment
-        # and the newline segments following it.
-        anchor_segment = context.segment
-        eof_newline_segments = []
-        for segment in complete_stack[::-1]:
-            if segment.is_type("newline"):
-                eof_newline_segments.append(segment)
-            elif segment.name not in ("whitespace", "Dedent"):
-                break
-            anchor_segment = segment
-
-        if len(eof_newline_segments) == 1:
+        # Include current segment for complete stack and reverse.
+        parent_stack: Segments = context.surrogates.parent_stack
+        raw_stack: Segments = context.surrogates.raw_stack
+        complete_stack = raw_stack.append(context.segment)
+        reversed_complete_stack = complete_stack.reversed()
+
+        # Find the trailing newline segments.
+        trailing_newlines = reversed_complete_stack.select(
+            select_if=["newline"], loop_while=[lambda x: x.is_whitespace or x.is_meta]
```

Comment:
Forgot to mention in my PR for this, but maybe this also highlights that we want `and()` & `or()` predicates as well, like we're suggesting with `not()`?

So we can write this line as:
```
select_if=["newline"], loop_while=[or(segpred.is_whitespace, segpred.is_meta)]
```

Reply:
WIll do!